pkg/storage/storage.go

```diff
@@ -61,66 +62,185 @@ func Remove(client *occlient.Client, name string, componentName string, applicat
+			if pvc.Name == "" {
+				return nil, fmt.Errorf("no PVC associated")
+			}
+			storageName := getStorageFromPVC(&pvc)
+			storageSize := pvc.Spec.Resources.Requests[corev1.ResourceStorage]
+			storagePath := mountedStorageMap[storageName]
 			storageList = append(storageList, StorageInfo{
-				Name: storage,
-				Size: size.String(),
+				Name: storageName,
+				Size: storageSize.String(),
+				Path: storagePath,
 			})
 		}
 	}
 	return storageList, nil
 }
 
+// Exists checks if the given storage exists in the given application
+func Exists(client *occlient.Client, storageName string, componentName string, applicationName string) (bool, error) {
+	var labels = make(map[string]string)
+	labels[applabels.ApplicationLabel] = applicationName
+	labels[storagelabels.StorageLabel] = storageName
+	selector := util.ConvertLabelsToSelector(labels)
+	pvcs, err := client.GetPVCsFromSelector(selector)
+	if err != nil {
+		return false, errors.Wrapf(err, "unable to list storage for component %v", componentName)
+	}
+
+	if len(pvcs) <= 0 {
+		return false, nil
+	}
+	return true, nil
+}
+
 // generatePVCNameFromStorageName generates a PVC name from the given storage
 // name
 func generatePVCNameFromStorageName(storage string) string {
 	return fmt.Sprintf("%v-pvc", storage)
 }
 
-// getStorageFromPVC returns the storage assocaited with the given PVC
-func getStorageFromPVC(pvc corev1.PersistentVolumeClaim) string {
+// getStorageFromPVC returns the storage associated with the given PVC
+func getStorageFromPVC(pvc *corev1.PersistentVolumeClaim) string {
 	if _, ok := pvc.Labels[storagelabels.StorageLabel]; !ok {
 		return ""
 	}
 	return pvc.Labels[storagelabels.StorageLabel]
 }
 
-// removeStorage removes the given PVC from the given Deployment Config and also
-// deletes the PVC
-func removeStorage(client *occlient.Client, pvc string, dc string) error {
-	// Remove PVC from Deployment Config
-	if err := client.RemoveVolumeFromDeploymentConfig(pvc, dc); err != nil {
-		return errors.Wrapf(err, "unable to remove volume: %v from Deployment Config: %v", pvc, dc)
+// getPVCNameFromStorageName returns the PVC associated with the given storage
+func getPVCNameFromStorageName(client *occlient.Client, storageName string) (string, error) {
+	var labels = make(map[string]string)
+	labels[storagelabels.StorageLabel] = storageName
+
+	selector := util.ConvertLabelsToSelector(labels)
+	pvcs, err := client.GetPVCNamesFromSelector(selector)
+	if err != nil {
+		return "", errors.Wrapf(err, "unable to get PVC names for selector %v", selector)
 	}
+	numPVCs := len(pvcs)
+	if numPVCs != 1 {
+		return "", fmt.Errorf("expected exactly one PVC attached to storage %v, but got %v, %v", storageName, numPVCs, pvcs)
+	}
+	return pvcs[0], nil
+}
 
-	// Delete PVC
-	if err := client.DeletePVC(pvc); err != nil {
-		return errors.Wrapf(err, "unable to delete PVC: %v", pvc)
+func getComponentAppNameFromStorageName(client *occlient.Client, storageName string) (string, error) {
```

Comment:
This is a really confusing name. :-(
From the code it looks like it is returning the Component name. Why the is function name mentions AppName?